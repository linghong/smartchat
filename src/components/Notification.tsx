import React, { FC } from 'react'

interface Props {
  type?: 'error' | 'loading' | 'success' | 'status';
  message: string | null;
}

const Notification: FC<Props> = ({ type = 'status', message }) => {
  if(message === null) return null
  let className = ''
  let role= ''

  switch (type) {
    case 'error':
      className = 'bold text-red-600'
      role = 'alert'
      break
    case 'loading':
      className = 'bold text-gray-600'
      role = 'status'
      break
    case 'success':
      className = 'bold text-green-600'
      role = 'status'
      break
    case 'status':
      role = 'status'
      className = 'bold text-yellow-600'
    default:
      role = ''
      className = ''
  }

  return <p className={className} role={role}>{message}</p>
}

export default Notification