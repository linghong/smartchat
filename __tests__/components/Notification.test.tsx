import React from 'react'
import { render, screen } from '@testing-library/react'
import '@testing-library/jest-dom'
import Notification from '@/src/components/Notification'

describe('Notification Component', () => {
  
  test('renders null when message is null', () => {
    const { container } = render(<Notification message={null} />)
    expect(container.firstChild).toBeNull()
  })

  test('renders with error type correctly', () => {
    render(<Notification type="error" message="Error message" />)
    const messageElement = screen.getByText('Error message')
    expect(messageElement).toHaveClass('bold', 'text-red-600')
    expect(messageElement).toHaveAttribute('role', 'alert')
    expect(messageElement).toHaveAttribute('aria-live', 'assertive')
  })

  test('renders with loading type correctly', () => {
    render(<Notification type="loading" message="Loading message" />)
    const messageElement = screen.getByText('Loading message')
    expect(messageElement).toHaveClass('bold', 'text-gray-600')
    expect(messageElement).toHaveAttribute('role', 'status')
    expect(messageElement).toHaveAttribute('aria-live', 'assertive')
  })

  test('renders with success type correctly', () => {
    render(<Notification type="success" message="Success message" />)
    const messageElement = screen.getByText('Success message')
    expect(messageElement).toHaveClass('bold', 'text-green-600')
    expect(messageElement).toHaveAttribute('role', 'status')
    expect(messageElement).toHaveAttribute('aria-live', 'polite')
  })

  test('renders with status type correctly', () => {
    render(<Notification type="status" message="Status message" />)
    const messageElement = screen.getByText('Status message')
    expect(messageElement).toHaveClass('bold', 'text-yellow-600')
    expect(messageElement).toHaveAttribute('role', 'status')
    expect(messageElement).toHaveAttribute('aria-live', 'polite')
  })

  test('renders with default type correctly', () => {
    render(<Notification message="Default message" />)
    const messageElement = screen.getByText('Default message')
    // Check for default behavior. 
    expect(messageElement).not.toHaveClass()
    expect(messageElement).toHaveAttribute('role', "")
    expect(messageElement).toHaveAttribute('aria-live', 'off')
  })

  test('Notification component snapshot', () => {
    const { asFragment } = render(<Notification message="Default message" />)

    expect(asFragment()).toMatchSnapshot()
  })
})