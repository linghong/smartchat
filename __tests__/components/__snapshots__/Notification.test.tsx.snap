// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification Component Notification component snapshot 1`] = `
<DocumentFragment>
  <div
    class="p-4"
  >
    <p
      aria-live="off"
      class=""
      role=""
    >
      Default message
    </p>
  </div>
</DocumentFragment>
`;
